codegen/src/api/runtime_apis.rs

```diff
@@ -63,21 +60,21 @@ fn generate_runtime_api(
             }
         );
 
-        let output = type_gen.resolve_type_path(method.output.id);
+        let output = type_gen.resolve_type_path(method.output_ty());
 
-        let Ok(call_hash) =
-            subxt_metadata::get_runtime_api_hash(metadata, trait_name, &method.name) else {
-                return Err(CodegenError::MissingRuntimeApiMetadata(
-                    trait_name.into(),
-                    method.name.clone(),
-                ))
-            };
+        let Some(call_hash) = api.method_hash(method.name()) else {
+            return Err(CodegenError::MissingRuntimeApiMetadata(
+                trait_name_str.to_owned(),
+                method_name_str.to_owned(),
+            ))
+        };
 
         let method = quote!(
             #docs
             pub fn #method_name(&self, #( #params, )* ) -> #crate_path::runtime_api::Payload<types::#struct_name, #output> {
                 #crate_path::runtime_api::Payload::new_static(
-                    #runtime_fn_name,
+                    #trait_name_str,
+                    #method_name_str,
```

Comment:
I moved Runtime APIs to be split like calls etc, so that you provide separately the trait name and then method name. One reason for doing this was simply allow consistency in the metadata between how Runtime APIs and Pallet based things are accessed (ie you ask for trait name and then can ask for a specific method or just get docs for the trait ot whatever.

Reply:
Issue opened for this: #976